var/spack/repos/spack_repo/builtin/packages/greenx/package.py

```diff
@@ -38,6 +41,8 @@ class Greenx(CMakePackage):
 
     depends_on("gmp", when="+gmp")
 
+    patch("cmake.patch", when="@2.1:")
```

Comment:
Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

Reply:
- The patch is needed because greenX creates a `greenXconfig.cmake` file (used by find_package) that does not include its target dependency. So when greenX is included in a project (in that case CP2K) it can lead to a linking error at the end. this patch fixes it.
- I wrote the patch